### PR TITLE
armv7-m/armv8-m: move up_pref* api to common place

### PR DESCRIPTION
**Pull request by anchao**
## Summary

armv7-m/armv8-m: move up_pref* api to common place

## Impact

N/A

## Testing

CI check

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/nuttx/arch.h [hunk 1]
```diff
@@ -2555,8 +2555,10 @@ void arch_sporadic_resume(FAR struct tcb_s *tcb);
  *
  *   The second interface simple converts an elapsed time into well known
  *   units.
+ *
  ****************************************************************************/
 
+void up_perf_init(FAR void *arg);
```
Comment (pkarashchenko) [inline include/nuttx/arch.h:2561]:
```suggestion
void up_perf_init(uintptr_t arg);
```
Just an option to have less type casts.
I'm fine with current way.
Or we can make it more explicit `void up_perf_init(uint32_t cpu_freq);`

**Review by pkarashchenko — APPROVED** [review]

File: include/nuttx/arch.h (hunk 1 above)
Comment (pkarashchenko) [reply include/nuttx/arch.h:2561]: Just feedback if you will apply suggestion, so I can either wait for update or merge it as is.

Comment (xiaoxiang781216) [reply include/nuttx/arch.h:2561]: It intend to use void * here, if other arch implementation may need more argument, he can define a struct to contain more info without modify the prototype.

Comment (pkarashchenko) [reply include/nuttx/arch.h:2561]: `uintptr_t` will give same effect. But let's keep `void *`

**Pull request merged**